Position position
uint8 id
string class